chapters/2-abstraction.md

```diff
@@ -0,0 +1,100 @@
+# Abstraction
+
+Functional abstraction allows us to treat functions as black boxes.
+We just care about what a function does and how to use it.
+
+This is helpful when implementing more complex programs.
+We use **Wishful Thinking** to imagine that we have functions
+which do what we want. 
+We start by thinking of what they should do and how to use them.
+This helps you to think about what functions and data you need
+before you implement the function.
+
+## Writing Stubs
+
+When writing a program, we can practise wishful thinking by writing
+**function stubs**.
+
+A [function stub](https://en.wikipedia.org/wiki/Method_stub)
+is a piece of code that stands in for the actual functionality.
+
+The stub allows us to get the main outline of the program up
+before figuring out the details.
+While incrementally working towards the final product,
+the program can be run without syntax errors, allowing us to write
+small manageable chunks at a time.
```

Comment:
Show, don't tell. haha. Maybe can give a very compelling stub example, then formalize that knowledge with the definitions, properties and benefits of stubs.